### PR TITLE
fix timestamp bug

### PR DESCRIPTION
**Pull request by xinyual**
### Description
This pr fix the timestamp bug when take time argument. It uses the current function properties. We now follow the same logic as https://github.com/opensearch-project/sql/blob/e23a61dcd9f15864a6bf4a5daa7ac4263a890716/core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunctions.java#L880 to firstly transfer it to timestamp.

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]
https://github.com/opensearch-project/sql/issues/3545

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
 - [ ] New functionality has javadoc added.
 - [ ] New functionality has a user manual doc added.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/sql/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: integ-test/src/test/java/org/opensearch/sql/calcite/standalone/CalcitePPLDateTimeBuiltinFunctionIT.java [hunk 1]
```diff
@@ -101,6 +101,20 @@ public void testDate() {
             Date.valueOf("1984-04-12")));
   }
 
+  @Test
+  public void testTimestampWithTimeInput() {
+    String utcTomorrow = LocalDate.now().plusDays(1).toString();
```
Comment (qianheng-aws) [inline integ-test/src/test/java/org/opensearch/sql/calcite/standalone/CalcitePPLDateTimeBuiltinFunctionIT.java:106]: There may be an edge case this test should be flakey when it was ran at the time point crossing a day.

I think it should use a fixed time via Hook.CURRENT_TIMESTAMP once changing to use the time properties in Calcite.

**Review by qianheng-aws — APPROVED** [review]

File: integ-test/src/test/java/org/opensearch/sql/calcite/standalone/CalcitePPLDateTimeBuiltinFunctionIT.java (hunk 1 above)
Comment (xinyual) [reply integ-test/src/test/java/org/opensearch/sql/calcite/standalone/CalcitePPLDateTimeBuiltinFunctionIT.java:106]: Currently we don't use the latest refactor framework so Hook wouldn't work. But after changing it, I will use Hook in this IT.

**Comment by qianheng-aws** [comment]
CI failed for java21, is this a known failure?

**Comment by qianheng-aws** [comment]
> CI failed for java21, is this a known failure?

Should be addressed by this fix: https://github.com/opensearch-project/sql/pull/3543

File: integ-test/src/test/java/org/opensearch/sql/calcite/standalone/CalcitePPLDateTimeBuiltinFunctionIT.java (hunk 1 above)
Comment (qianheng-aws) [reply integ-test/src/test/java/org/opensearch/sql/calcite/standalone/CalcitePPLDateTimeBuiltinFunctionIT.java:106]: Please add a TODO here.

**Comment by LantaoJin** [comment]
Could you open an issue?

File: core/src/main/java/org/opensearch/sql/calcite/udf/datetimeUDF/TimestampFunction.java [hunk 2]
```diff
@@ -20,7 +19,8 @@
 /**
  * We need to write our own since we are actually implement timestamp add here
  * (STRING/DATE/TIME/DATETIME/TIMESTAMP) -> TIMESTAMP (STRING/DATE/TIME/DATETIME/TIMESTAMP,
- * STRING/DATE/TIME/DATETIME/TIMESTAMP) -> TIMESTAMP
+ * STRING/DATE/TIME/DATETIME/TIMESTAMP) -> TIMESTAMP It's implicitly transferred into timestamp, so
```
Comment (LantaoJin) [inline core/src/main/java/org/opensearch/sql/calcite/udf/datetimeUDF/TimestampFunction.java:22]: please format this java doc for more readable

File: integ-test/src/test/java/org/opensearch/sql/calcite/standalone/CalcitePPLDateTimeBuiltinFunctionIT.java (hunk 1 above)
Comment (penghuo) [reply integ-test/src/test/java/org/opensearch/sql/calcite/standalone/CalcitePPLDateTimeBuiltinFunctionIT.java:107]: > `timestamp(time('12:00:00'), time('12:00:00'))`

what is meaning of time+time? should we limit the first argument be date only?

File: core/src/main/java/org/opensearch/sql/calcite/udf/datetimeUDF/TimestampFunction.java (hunk 2 above)
Comment (xinyual) [reply core/src/main/java/org/opensearch/sql/calcite/udf/datetimeUDF/TimestampFunction.java]: Done.

File: integ-test/src/test/java/org/opensearch/sql/calcite/standalone/CalcitePPLDateTimeBuiltinFunctionIT.java (hunk 1 above)
Comment (xinyual) [reply integ-test/src/test/java/org/opensearch/sql/calcite/standalone/CalcitePPLDateTimeBuiltinFunctionIT.java:107]: In v2, we will implicitly convert the time into timestamp. The logic of v2 timestamp is transfer two arguments into timestamp (if we have two.), then extract time part from the second timestamp and then add it to the first timestamp.

**Comment by xinyual** [comment]
> Could you open an issue?

Done. https://github.com/opensearch-project/sql/issues/3545

**Review by LantaoJin — APPROVED** [review]

**Review by qianheng-aws — APPROVED** [review]
